File: src/jkorn/pvpcore/commands/EditWorldCommand.php
<?php
/**
 * Created by PhpStorm.
 * User: jkorn2324
 * Date: 2020-06-26
 * Time: 17:50
 */

declare(strict_types=1);

namespace jkorn\pvpcore\commands;


use jkorn\pvpcore\forms\PvPCoreForms;
use jkorn\pvpcore\PvPCore;
use jkorn\pvpcore\utils\Utils;
use pocketmine\command\Command;
use pocketmine\command\CommandSender;
use pocketmine\command\utils\CommandException;
use pocketmine\Player;
use pocketmine\utils\TextFormat;

class EditWorldCommand extends Command
{

    public function __construct()
    {
        parent::__construct("editWorld", "Sends the edit world Knockback form to the player.", "Usage: /editWorld", ["editworld", "editworldkb"]);
        parent::setPermission("pvpcore.permission.edit");
    }

    /**
     * @param CommandSender $sender
     * @param string $commandLabel
     * @param string[] $args
     *
     * @return mixed
     * @throws CommandException
     */
    public function execute(CommandSender $sender, string $commandLabel, array $args)
    {
        if(!$sender instanceof Player)
        {
            $sender->sendMessage(Utils::getPrefix() . TextFormat::RED . " Console can't use this command.");
            return true;
        }

        if(!$this->testPermission($sender))
        {
            return true;
        }

        $form = PvPCoreForms::getWorldMenu(
            $sender,
            PvPCore::getWorldHandler()->getWorld($sender->getLevel()),
            false
        );

        $sender->sendForm($form);
        return true;
    }
}

File: Nukkit/src/pvpcore/commands/EditWorldCommand.java
package pvpcore.commands;

import cn.nukkit.Player;
import cn.nukkit.command.Command;
import cn.nukkit.command.CommandSender;
import cn.nukkit.form.window.FormWindow;
import cn.nukkit.utils.TextFormat;
import pvpcore.PvPCore;
import pvpcore.forms.PvPCoreForms;
import pvpcore.utils.Utils;
import pvpcore.worlds.PvPCWorld;

public class EditWorldCommand extends Command
{

    public EditWorldCommand()
    {
        super("editWorld", "Sends the edit world Knockback form to the player.", "Usage: /editWorld", new String[]{"editworld", "editworldkb"});
        super.setPermission("pvpcore.permission.edit");
    }

    /**
     * Executes the command.
     * @param commandSender - The command sender.
     * @param s - The command label.
     * @param strings - The arguments for the commands.
     * @return true if command executes successfully, false otherwise.
     */
    @Override
    public boolean execute(CommandSender commandSender, String s, String[] strings)
    {
        if(!(commandSender instanceof Player))
        {
            commandSender.sendMessage(Utils.getPrefix() + TextFormat.RED.toString() + " Console can't use this command.");
            return true;
        }

        if(!this.testPermission(commandSender))
        {
            return true;
        }

        PvPCWorld world = PvPCore.getWorldHandler().getWorld(((Player) commandSender).getLevel());
        System.out.println(world);

        FormWindow window = PvPCoreForms.getWorldMenu(
                (Player)commandSender,
                world,
                false
        );
        ((Player) commandSender).showFormWindow(window);
        return true;
    }
}
